### PR TITLE
Catch-all page rendering Routes ↔ Landing Pages

### PR DESCRIPTION
**Pull request by j0nas**
# Catch-all page rendering Routes ↔ Landing Pages

## Intent

- Add a `[[...slug].tsx` catch-all page which renders Sanity _Route_ entities and their respective _Landing Page_ entities
- Extracts `CookieConsent` component and rewrites styling in plain CSS
- Deletes TailwindCSS and all "legacy" Tailwind-based pages, removes Tailwind-related dependencies and configuration

## Description

This PR is the result of a [request](https://wja.slack.com/archives/C01CT37P5HC/p1644823800644089) by @kmelve on Slack, calling for replicating the Route ↔ Landing Page separation pattern referenced to in [this Loom](https://www.loom.com/share/49d2263fcd4341e6aca313efbab576e7) and as seen in e.g. the www-sanity-io repo.

Other files defined in _/pages_ will have precedence over the new catch-all route, so e.g. [/about](http://localhost:3000/about) will still render _/web/pages/about.tsx_. This is intentional, both due to the fact that Sanity is currently lacking the corresponding [data for Routes](http://localhost:3333/desk/route), and because <em>Landing Page</em>s seem to only support regular "sections" (as typed out by _/web/types/Section.ts_) rather than the more complex and varying nature of the content in the current routes (<em>/about, /program, /tickets, etc</em>) as required by the wireframes.

Note that this breaks `staging`'s backward compatibility with `main`; after this PR is merged to `staging`, `staging` will no longer be mergable to `main` without breaking the [current home page](https://structured-content-2022.sanity.build/). Having `staging` be our development branch (and only merging it to `main` at the time of the new page's release in its entirety) was the plan all along, but this PR breaks a backward compatibility which `staging` has had until now.
 
## Testing this PR

1. `docker-compose up` in the repository's root
2. Go to [/test](http://localhost:3000/test) and verify that the _Page_ associated with the [test route](http://localhost:3333/desk/route;188f7c30-b8c4-4c46-9917-33722517f9ce) I created is rendered.
3. Go to [/home](http://localhost:3000/home) (and the other pages linked in the nav) and verify that they all still look as awful as they did prior to this PR
4. Open any page in an Incognito window and verify that you get the `<CookieConsent />` banner rendered. Verify that the banner looks and behaves exactly like that in the current ["production"](https://structured-content-2022.sanity.build/) version

### CODE REVIEW

**Review by oyvind-stenhaug — CHANGES_REQUESTED** [review]
Regarding step 5 of the "Testing this PR" section, I noticed the following minor differences:

- Doesn't give the buttons equal width on narrow screens. Instead, all the free space seems to be given to "No, thanks" , while "OK" remains at some minimum or intrinsic width.
- No hover effect (in the original, the link text and button backgrounds become lighter)
- No `outline: auto` on `:focus-visible` (which the original seems to have)… this seems to make a difference for the link outline in Firefox, at least on my install. (I actually  wasn't aware that it behaves like this! The user-agent styles give a 1px dotted outline. But it renders `auto` differently, with a more visible indicator)

Feel free to treat as nit, maybe we'll revisit the styles here regardless…

File: web/components/CookieConsent/CookieConsent.module.css [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  }
+}
+
+@media (min-width: 768px) {
+  .consentContent {
+    max-width: 28rem;
+  }
+}
+
+.consentButtonWrapper {
```
[the remark this answers is not in the document]
Comment (j0nas) [reply web/components/CookieConsent/CookieConsent.module.css]: Removed all redundant `consent` prefixes from class names in [a9d1a80](https://github.com/sanity-io/structured-content-2022/pull/9/commits/a9d1a80063e8cb40fa681957f666d54c7b66da83).

**Comment by j0nas** [comment]
> Regarding step 5 of the "Testing this PR" section, I noticed the following minor differences:

> * Doesn't give the buttons equal width on narrow screens. Instead, all the free space seems to be given to "No, thanks" , while "OK" remains at some minimum or intrinsic width.
> * No hover effect (in the original, the link text and button backgrounds become lighter)
> * No outline: auto on :focus-visible (which the original seems to have)… this seems to make a difference for the link outline in Firefox, at least on my install. (I actually wasn't aware that it behaves like this! The user-agent styles give a 1px dotted outline. But it renders auto differently, with a more visible indicator)

Fixed all of these in 9839fa8. Good catch regarding the weird spacing on small viewports!

**Review by oyvind-stenhaug — APPROVED** [review]
(Step 2 of the "Testing" section no longer applies, because a route has since been added for "/".)

Nit: Hover style for the link is still not copied over.

**Comment by j0nas** [comment]
@oyvind-stenhaug Good catch, once again! Added the missing `:hover` style in b8ad0a2, and updated the README.

**Pull request merged**